cloud: add private link connection docs (#22192)
pingcap/docs: tidb-cloud/serverless-private-link-connection-to-aws-confluent.md

> # Set up a Private Link Connection to AWS Confleunt

Reviewer: ```suggestion
# Set up a Private Link Connection to AWS Confluent
```

> **Note**
>
> Only Confluent Dedicated clusters are supported.

> This document describes how to connect to an AWS Confluent Dedicated Cluster using an AWS Endpoint Service private link connection.

Reviewer: ```suggestion
This document describes how to connect to an AWS Confluent Dedicated cluster using an AWS Endpoint Service private link connection.
```

    1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the cluster overview page of the TiDB cluster, and then click **Settings** > **Networking** in the left navigation pane.
    2. On the **Private Link Connection For Dataflow**, click **Create Private Link Connection**.
    3. You can find the AWS account ID and available zones information.

## Step 1. Set up a Confluent Cloud network

Identify a Confluent Cloud network you want to use, or [set up a new Confluent Cloud network](https://docs.confluent.io/cloud/current/networking/ccloud-network/aws.html#create-ccloud-network-aws).

The Confluent Cloud network must meet the following requirements:

- The network must be in the same AWS region as your {{{ .essential }}}.
- The network must have overlapping availability zones as your {{{ .essential }}}, recommended to have same availability zones.

On the `Network overview` page, obtain the `DNS subdomain` of the Confluent Cloud network. You need to extract the unique name from it.
For example, if the `DNS subdomain` is `use1-az1.domnprzqrog.us-east-1.aws.confluent.cloud`, then the unique name is `domnprzqrog.us-east-1`.

> Please save the unique name for later use.

Reviewer: ```suggestion

Save the unique name for later use.
```

> **Note**
>
> The Confluent Cloud Dedicated cluster you want TiDB Cloud to connect to must under this network.

## Step 2. Add a PrivateLink Access to the network

> Please add a PrivateLink Access to the network you identified or set up in Step 1. Refer to [Add a PrivateLink Access in Confluent Cloud](https://docs.confluent.io/cloud/current/networking/private-links/aws-privatelink.html#add-a-privatelink-access-in-ccloud).

Reviewer: ```suggestion
Add a PrivateLink Access to the network you identified or set up in Step 1. Refer to [Add a PrivateLink Access in Confluent Cloud](https://docs.confluent.io/cloud/current/networking/private-links/aws-privatelink.html#add-a-privatelink-access-in-ccloud).
```

---
title: Connect to AWS Confluent via a Private Link Connection

> summary: Learn how to connect to an AWS Confluent instance using an AWS Confluent Endpoint Service private link connection.

Reviewer 2: ```suggestion
summary: Learn how to connect to an AWS Confluent instance using an AWS Endpoint Service private link connection.
```

---

# Connect to Confluent Cloud via a Private Link Connection

> This document describes how to connect to a Confluent Cloud Dedicated cluster on AWS using an AWS Endpoint Service private link connection.

Reviewer 2: ```suggestion
This document describes how to connect a {{{ .essential }}} cluster to a [Confluent Cloud Dedicated cluster](https://docs.confluent.io/cloud/current/clusters/cluster-types.html) on AWS using an [AWS Endpoint Service private link connection](/tidb-cloud/serverless-private-link-connection.md).
```

> **Note**
>

> > - Only Confluent Cloud Dedicated clusters on AWS are supported.

Reviewer 2: ```suggestion
> Only Confluent Cloud Dedicated clusters on AWS are supported for private link connections.
```

## Prerequisites
   
- You have a Confluent Cloud account.

- Confirm that your {{{ .essential }}} is active in AWS. Retrieve and save the following details for later use:

>     - Account ID

Reviewer 2: ```suggestion
    - AWS Account ID
```

    - Availability Zones (AZ)

To view the the AWS account ID and available zones, do the following:

1. In the [TiDB Cloud console](https://tidbcloud.com), navigate to the cluster overview page of the TiDB cluster, and then click **Settings** > **Networking** in the left navigation pane.

> 2. On the **Private Link Connection For Dataflow**, click **Create Private Link Connection**.

Reviewer 2: ```suggestion
2. In the **Private Link Connection For Dataflow** area, click **Create Private Link Connection**.
```

3. You can find the AWS account ID and available zones.

## Step 1. Set up a Confluent Cloud network

Identify a Confluent Cloud network that you want to use, or [create a new Confluent Cloud network on AWS](https://docs.confluent.io/cloud/current/networking/ccloud-network/aws.html#create-ccloud-network-aws).

The Confluent Cloud network must meet the following requirements:

- Type: the network must be a **PrivateLink** network.
- Region match: the instance must reside in the same AWS region as your {{{ .essential }}} cluster.

> - AZ (Availability Zone) availability: the availability zones must overlap with those of your {{{ .essential }}} cluster.

Reviewer 2: ```suggestion
- AZ (Availability Zone) availability: the availability zones of the network must overlap with those of your {{{ .essential }}} cluster.
```

To get the unique name of the Confluent Cloud network:

> 

Reviewer 2: Please specify where to perform the following steps.

Author: yes，it is on the Confluent Cloud console. have added this

> 1. On the `Network overview` page, obtain the `DNS subdomain` of the Confluent Cloud network. 

Reviewer 2: ```suggestion
1. On the **Network overview** page, obtain the DNS subdomain of the Confluent Cloud network. 
```

2. Extract the unique name from it. For example, if the `DNS subdomain` is `use1-az1.domnprzqrog.us-east-1.aws.confluent.cloud`, then the unique name is `domnprzqrog.us-east-1`.
3. Save the unique name for later use.

> **Note**
>

> > The Confluent Cloud Dedicated cluster must be deployed in this network.

Author: have added this

## Step 2. Add a PrivateLink Access to the network

Add a PrivateLink Access to the network you identified or set up in [Step 1](#step-1-set-up-a-confluent-cloud-network). For more information, see [Add a PrivateLink Access in Confluent Cloud](https://docs.confluent.io/cloud/current/networking/private-links/aws-privatelink.html#add-a-privatelink-access-in-ccloud).

During the process, you need to:

- Provide the TiDB Cloud AWS account ID that you obtain in [Prerequisites](#prerequisites).

> - Save the `VPC Service Endpoint` provided by Confluent Cloud for later use, usually in the format of `com.amazonaws.vpce.<region>.vpce-svc-xxxxxxxxxxxxxxxxx`.

Reviewer 2: ```suggestion
- Save the `VPC Service Endpoint` provided by Confluent Cloud for later use, usually in the `com.amazonaws.vpce.<region>.vpce-svc-xxxxxxxxxxxxxxxxx` format.
```

## Step 3. Create a private link connection in TiDB Cloud

To create a private link connection in TiDB Cloud, do the following:

### 1. Create the AWS Endpoint Service private link connection

For more information, see [Create an AWS Endpoint Service private link connection](/tidb-cloud/serverless-private-link-connection.md#create-an-aws-endpoint-service-private-link-connection).

> **Note:**
>
> For Confluent Cloud Dedicated clusters on AWS, you do not need to go to the detail page of your endpoint service on the AWS console to accept the endpoint connection request from TiDB Cloud.

### 2. Attach domains to the private link connection

> 

Reviewer 2: ```suggestion

Attach the Confluent Cloud service domains to the private link connection so that dataflow services in TiDB Cloud can access the Confluent cluster.

```

3. In the displayed dialog, you can find the AWS account ID and available zones.

## Step 1. Set up a Confluent Cloud network

Identify a Confluent Cloud network that you want to use, or [create a new Confluent Cloud network on AWS](https://docs.confluent.io/cloud/current/networking/ccloud-network/aws.html#create-ccloud-network-aws).

The Confluent Cloud network must meet the following requirements:

- Type: the network must be a **PrivateLink** network.
- Region match: the network must reside in the same AWS region as your {{{ .essential }}} cluster.
- AZ (Availability Zone) availability: the availability zones of the network must overlap with those of your {{{ .essential }}} cluster.

To get the unique name of the Confluent Cloud network:

1. Go to the **Environments** where your Confluent Cloud network is located on the [Confluent Cloud Console](https://confluent.cloud/environments).
2. Click **Network management** and choose **For dedicated clusters** to find the network you created.
3. Go to the **Network overview** page to obtain the DNS subdomain of the Confluent Cloud network. 

> 4. Extract the unique nae from the DNS subdomain. For example, if the DNS subdomain is `use1-az1.domnprzqrog.us-east-1.aws.confluent.cloud`, then the unique name is `domnprzqrog.us-east-1`.

Reviewer 2: ```suggestion
4. Extract the unique name of your Confluent Cloud network from the DNS subdomain. For example, if the DNS subdomain is `use1-az1.domnprzqrog.us-east-1.aws.confluent.cloud`, then the unique name is `domnprzqrog.us-east-1`.
```

5. Save the unique name for later use.

> **Note**
>
> The Confluent Cloud Dedicated cluster must be deployed in this network.

## Step 2. Add a PrivateLink Access to the network

Add a PrivateLink Access to the network you identified or set up in [Step 1](#step-1-set-up-a-confluent-cloud-network). For more information, see [Add a PrivateLink Access in Confluent Cloud](https://docs.confluent.io/cloud/current/networking/private-links/aws-privatelink.html#add-a-privatelink-access-in-ccloud).

During the process, you need to:

- Provide the TiDB Cloud AWS account ID that you obtain in [Prerequisites](#prerequisites).
- Save the `VPC Service Endpoint` provided by Confluent Cloud for later use, usually in the `com.amazonaws.vpce.<region>.vpce-svc-xxxxxxxxxxxxxxxxx` format.

## Step 3. Create a Confluent Cloud dedicated cluster under the network

Please create a Confluent Cloud Dedicated cluster under the network you set up in [Step 1](#step-1-set-up-a-confluent-cloud-network). For more information, see [Create a dedicated cluster in Confluent Cloud](https://docs.confluent.io/cloud/current/clusters/create-cluster.html#create-ak-clusters).

> When creating the cluster, use existing Confluent Cloud network you set up in [Step 1](#step-1-set-up-a-confluent-cloud-network).

Reviewer 2: ```suggestion
Create a Confluent Cloud Dedicated cluster under the network you set up in [Step 1](#step-1-set-up-a-confluent-cloud-network). For more information, see [Create a dedicated cluster in Confluent Cloud](https://docs.confluent.io/cloud/current/clusters/create-cluster.html#create-ak-clusters).
```

## Step 4. Create a private link connection in TiDB Cloud

To create a private link connection in TiDB Cloud, do the following:

### 1. Create the AWS Endpoint Service private link connection

Create a private link connection in TiDB Cloud using the `VPC Service Endpoint` from Confluent Cloud.

For more information, see [Create an AWS Endpoint Service private link connection](/tidb-cloud/serverless-private-link-connection.md#create-an-aws-endpoint-service-private-link-connection).

> **Note:**
>
> For Confluent Cloud Dedicated clusters on AWS, you do not need to go to the detail page of your endpoint service on the AWS console to manually accept the endpoint connection request from TiDB Cloud.

### 2. Attach domains to the private link connection

Attach the Confluent Cloud service domains to the private link connection so that dataflow services in TiDB Cloud can access the Confluent cluster.

> For more information, see [Attach domains to a private link connection](/tidb-cloud/serverless-private-link-connection.md#attach-domains-to-a-private-link-connection).

Reviewer 2: ```suggestion
1. Create a private link connection in TiDB Cloud using the `VPC Service Endpoint` from Confluent Cloud.

    For more information, see [Create an AWS Endpoint Service private link connection](/tidb-cloud/serverless-private-link-connection.md#create-an-aws-endpoint-service-private-link-connection).

    > **Note:**
    >
    > For Confluent Cloud Dedicated clusters on AWS, you do not need to go to the detail page of your endpoint service on the AWS console to manually accept the endpoint connection request from TiDB Cloud.

2. Attach the Confluent Cloud service domains to the private link connection so that dataflow services in TiDB Cloud can access the Confluent cluster.

    For more information, see [Attach domains to a private link connection](/tidb-cloud/serverless-private-link-connection.md#attach-domains-to-a-private-link-connection).
```
